kernel/syscall.c

```diff
@@ -2671,8 +2671,26 @@ long myst_syscall_getsockname(
     myst_fdtable_t* fdtable = myst_fdtable_current();
     myst_sockdev_t* sd;
     myst_sock_t* sock;
+    myst_fdtable_type_t type;
+    void* device = NULL;
+    void* object = NULL;
+
+    if (!myst_is_good_addr(addrlen) || !myst_is_good_addr(addr))
+    {
+        ret = -EFAULT;
```

Comment:
use ERAISE(..)

Reply:
using ERAISE causes the tests to all fail - is there a way to do this other than ERAISE(EFAULT);?
```
getsockname01    1  TPASS  :  bad file descriptor successful
getsockname01    2  TFAIL  :  getsockname01.c:127: bad file descriptor ; returned -1 (expected -1), errno 2 (expected 88)
getsockname01    3  TFAIL  :  getsockname01.c:127: invalid socket buffer ; returned -1 (expected -1), errno 22 (expected 14)
getsockname01    4  TFAIL  :  getsockname01.c:127: invalid aligned salen ; returned -1 (expected -1), errno 22 (expected 14)
```

Reply:
Added, thank you